### DISCUSSION TITLE
Developer Getting Started

### DISCUSSION THREAD

**Discussion by nramsbottom** [MoleMan1024/audiowagon#123] [Q&A] [answered]
I'm trying to setup a development environment and have followed the instructions [here](https://moleman1024.github.io/audiowagon/developers.html) and so far things have worked out.

I had some build issues related to "Firebase" which required that I create a Firebase project and download a settings file. Things seemed to all work after fixing that.

I created a 1GB FAT32 formatted disk image with a bunch of audio files that I know to work with Audiowagon in my vehicle. I did this using dd and mkfs on Linux and have verified that it is a healthy image (see below.)
![image](https://github.com/MoleMan1024/audiowagon/assets/136175/956e83cc-0e8b-4f13-abc4-126310c13c69)

I fired up the emulator and launched the app.
![image](https://github.com/MoleMan1024/audiowagon/assets/136175/338f3842-72ac-4a38-9f23-4c74204302cc)

My problem is that the USB/SD Card is always considered to be "Not Connected", despite being visible in the file system (via Android Studio) and other applications, such as ES File Explorer.
![image](https://github.com/MoleMan1024/audiowagon/assets/136175/8169a28f-bd0a-4593-b01c-d489972a5d75)
![image](https://github.com/MoleMan1024/audiowagon/assets/136175/8ea0b1b7-e4d1-4e56-8734-064a5a196f53)

Did I miss something during the setup here? Why won't AW read the virtual SD Card?

**Comment by nramsbottom** [comment]
I see this comment...

```
/**
 * NOTE: SD card support is only enabled in debug builds used in the Android emulator
 */
```
but my APK is entitled `automotive-debug.apk`, which would suggest that it is running a debug build.

**Comment by nramsbottom** [comment]
I tried the `emulatorSDCard` build variant with no luck...

![image](https://github.com/MoleMan1024/audiowagon/assets/136175/517fc249-ab3f-4f75-8598-26bff34115a9)

**Reply by nramsbottom** [reply]
![image](https://github.com/MoleMan1024/audiowagon/assets/136175/a39666fa-1412-49aa-b39f-bf8a286f1f4f)

**Reply by nramsbottom** [reply]
Updating the code to use the GUID of my disk image (I guess it's generated somewhere?) and recreating the image to include a `many_files` directory yields some changes in the UI, but still no audio files visible in AW.

![image](https://github.com/MoleMan1024/audiowagon/assets/136175/1b70de44-172d-4975-9205-316d2018d5ea)

**Comment by MoleMan1024** [comment, marked as answer]
> Updating the code to use the GUID of my disk image

That is the correct approach.

In one of your screenshots it tries to load the "many_files" directory, that is part of my SD card image I use for testing. It is hardcoded, you can change that in https://github.com/MoleMan1024/audiowagon/blob/master/automotive/src/main/java/de/moleman1024/audiowagon/filestorage/AudioFileStorage.kt#L140

SD card will also need READ_EXTERNAL_STORAGE permission, it should ask for that when entering the settings. Make sure permission is granted.

Also try to switch to a different audio app and back to AudioWagon when you have set it to "Read metadata: when USB connected". Since there is no USB, the connection/disconnection event will never come. But restarting the app will trigger it as well (or index metadata manually).

**Reply by nramsbottom** [reply]
Dropping the many_files path worked. I was trying to make it work with as few code changes as possible, but I can live with this :)

![image](https://github.com/MoleMan1024/audiowagon/assets/136175/a1ea1afe-f79e-484c-b818-1659dcaebfec)
